plugin/PDFView.qml

```diff
@@ -214,17 +220,29 @@ SilicaFlickable {
                 }
 
                 canvas: pdfCanvas
+                selection: pdfSelection
+
+                onPositionChanged: contextHook.positionChanged(Qt.point(at.x - contentX,
+                                                                        at.y - contentY))
+
                 onLinkClicked: base.linkClicked(linkTarget, contextHook)
                 onGotoClicked: base.goToPage(page - 1, top, left,
                                              Theme.paddingLarge, Theme.paddingLarge)
-                onClicked: {
-                    if (selection.text.length > 0) {
-                        selection.unselect()
-                    } else {
-                        base.clicked()
-                    }
+                onSelectionClicked: base.selectionClicked(selection, contextHook)
+                onAnnotationClicked: base.annotationClicked(annotation, contextHook)
+                onTextAnnotationClicked: base.annotationClicked(annotation, contextHook)
+                onCaretAnnotationClicked: base.annotationClicked(annotation, contextHook)
+                onHighlightAnnotationClicked: base.annotationClicked(annotation, contextHook)
+                onClicked: base.clicked()
+                onAnnotationLongPress: base.annotationLongPress(annotation, contextHook)
+                onTextAnnotationLongPress: base.annotationLongPress(annotation, contextHook)
+                onCaretAnnotationLongPress: base.annotationLongPress(annotation, contextHook)
+                onHighlightAnnotationLongPress: base.annotationLongPress(annotation, contextHook)
```

Comment:
Ok, I will retry after lunch to use the unified signal emission from pdflinkarea and report if I stumble on the same issue as before.